jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -576,3 +576,7 @@ def vue_register_spectrum(self, event):
 
         self.add_results.add_results_from_plugin(spectrum)
         self._set_default_results_label()
+
+        # Link the result spectrum to the reference data of the spectrum viewer
+        self.app._link_new_data(self.app.get_viewer('spectrum-viewer').state.reference_data.label,
+                                self.results_label)
```

Comment:
is this redundant now with the logic in `app.py` or does it need to stay in?

Reply:
Ah, good catch!